parachain/primitives/ethereum/src/lib.rs

```diff
@@ -27,3 +39,182 @@ impl From<ethabi::Error> for DecodeError {
 		DecodeError::InvalidABI(err)
 	}
 }
+
+/// Complete block header id.
+#[derive(Clone, Copy, Default, Encode, Decode, PartialEq, RuntimeDebug)]
+pub struct HeaderId {
+	/// Header number.
+	pub number: u64,
+	/// Header hash.
+	pub hash: H256,
+}
+
+/// An Ethereum block header.
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
```

Comment:
Is using std features an issue here? Why do we need them?

Reply:
We don't strictly need `Serialize` and `Deserialize` afaik, but it might be convenient for development to load headers from e.g. JSON. It's mostly here because I copied the definition from [parity-bridges-common](https://github.com/paritytech/parity-bridges-common/blob/master/primitives/ethereum-poa/src/lib.rs) though.

W.r.t. the use of std features... are you familiar with Substrate's use of "std" and "no_std" to differentiate a normal build and WASM build? serde cannot be compiled to WASM. So we exclude its use in the WASM build. I'm not sure if you're familiar with how Substrate uses WASM. Let me know if you want more info

Reply:
I suppose the `derive(Serialize, Deserialize)` could be useful for unit tests which definitely do have `std` available.